keys.js

```diff
@@ -76,5 +77,37 @@ define(function(require) {
 		}
 	}
 
+	/**
+	 * Resolve all key-value pairs in the supplied object and return a promise
+	 * that will always fulfill with the outcome states of all input promises.
+	 * @param {object} object whose key-value pairs will be settled
+	 * @returns {Promise} promise for an object with the mapped and fully
+	 *  settled key-value pairs
+	 */
+	 function settle(object) {
+			var p = Promise._defer();
+			var resolver = Promise._handler(p);
+
+			var results = {};
+			var keys = Object.keys(object);
+			var promises = keys.map(function(k) { return object[k]; });
+			var pending = keys.length;
+
+			when.settle(promises).then(function(d) {
+				for(var i=0; i<keys.length; i++) {
+					results[keys[i]] = d[i];
+					if(--pending === 0) {
+						resolver.resolve(results);
+					}
+				}
+			});
```

Comment:
Cool, I wouldn't worry too much about trying to reduce the number of passes.  If you come up with something clever, great, but I def don't consider it a blocker for this PR :)